pkg/kubefed2/disable.go

```diff
@@ -148,48 +151,104 @@ func DisableFederation(cmdOut io.Writer, config *rest.Config, typeConfigName ctl
 	if err != nil {
 		return errors.Wrap(err, "Failed to get federation clientset")
 	}
+
+	write := func(data string) {
+		if cmdOut != nil {
+			cmdOut.Write([]byte(data))
+		}
+	}
+
 	typeConfig := &fedv1a1.FederatedTypeConfig{}
-	err = client.Get(context.TODO(), typeConfig, typeConfigName.Namespace, typeConfigName.Name)
+	ftcExists, err := checkFederatedTypeConfigExists(client, typeConfig, typeConfigName, write)
 	if err != nil {
-		return errors.Wrapf(err, "Error retrieving FederatedTypeConfig %q", typeConfigName)
+		return err
 	}
 
 	if dryRun {
 		return nil
 	}
 
-	write := func(data string) {
-		if cmdOut != nil {
-			cmdOut.Write([]byte(data))
+	if ftcExists {
+		if typeConfig.Spec.PropagationEnabled {
+			err = disablePropagationController(client, typeConfig, typeConfigName, write)
+			if err != nil {
+				return err
+			}
+		}
+		checkPropagationControllerStopped(client, typeConfigName, write)
+		err = deleteFederatedTypeConfig(client, typeConfig, typeConfigName, write)
+		if err != nil {
+			return err
 		}
 	}
 
+	if delete {
+		err = deleteFederatedType(config, typeConfig, write)
+		if err != nil {
+			return err
+		}
+	}
+
+	return nil
+}
+
+func checkFederatedTypeConfigExists(client genericclient.Client, typeConfig *fedv1a1.FederatedTypeConfig, typeConfigName ctlutil.QualifiedName, write func(string)) (bool, error) {
+	err := client.Get(context.TODO(), typeConfig, typeConfigName.Namespace, typeConfigName.Name)
+	if err == nil {
+		return true, nil
+	}
+
+	if apierrors.IsNotFound(err) {
+		write(fmt.Sprintf("FederatedTypeConfig %q already removed\n", typeConfigName))
+		return false, nil
+	}
+
+	return false, errors.Wrapf(err, "Error retrieving FederatedTypeConfig %q", typeConfigName)
+}
+
+func disablePropagationController(client genericclient.Client, typeConfig *fedv1a1.FederatedTypeConfig, typeConfigName ctlutil.QualifiedName, write func(string)) error {
```

Comment:
Isn't this `stopPropagationController`, meaning the controller for this type would actually be stopped; right?